Issue 1549: disable parallel xargs where not supported (i.e. on busybox)

gonewest818 opened the pull request:
Fixes #1549 

Summary of changes:
- in scripts/build_plugins.sh we now check if this is a busybox system
- if so, don't pass the "-P" flag to xargs, because busybox doesn't support that flag
- also change the logging output to show we're running serially

Testing done:
- performed a make in alpine linux

@intelsdi-x/snap-maintainers

andrzej-k commented on scripts/build_plugins.sh:
```diff
@@ -51,12 +51,21 @@ else
 p=${BUILD_JOBS:-"${p}"}
 
+# disable parallel xargs where not supported (ie on busybox)
+if [[ ! -x "$(command -v readlink)" || "$(readlink $(which xargs))" == "/bin/busybox" ]]; then
```
Not sure, but shouldn't this be ```"$(readlink -f $(which xargs))"``` (added ```-f```)?

gonewest818 replied:
OS X doesn't support that -f flag, and I was trying to keep the check simple by not having additional layers of conditionals based on OS.

What I can do is check for busybox only in the Linux case, not the Darwin case. That's probably a safe assumption (have you ever seen a Darwin with busybox ?)

andrzej-k replied:
I'm fine with supporting only Linux case. 
@jcooklin: Are you aware if busybox is supported on Darwin?

gonewest818 commented:
I just pushed a change to my branch that addresses the comment about `readlink -f`, but as discussed, performing this check only when `[ "${OS}" == "Linux" ]` .
